### ISSUE TITLE
Style/RedundantStringEscape breaks with a symbol with double quotes

### ISSUE THREAD

**Issue by robotdana** [rubocop/rubocop#11962]
a symbol with double quotes and an interpolation (so it requires double quotes) and an escaped literal double quote (required to be escaped because it matches the quotes) breaks the Style/RedundantStringEscape cop

--------

## Expected behavior

```ruby
value = 1
:"\"#{value}\""
```

would be left alone by rubocop. i have my reasons for writing something this cursed

## Actual behavior

```
$ rubocop -a test.rb --only Style/RedundantStringEscape --debug
For /Users/dana/Projects: Default configuration from /Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /Users/dana/Projects/test.rb
An error occurred while Style/RedundantStringEscape cop was inspecting /Users/dana/Projects/test.rb:4:7.
undefined method `source' for nil:NilClass

          delimiters = [node.loc.begin.source[-1], node.loc.end.source[0]]
                                      ^^^^^^^
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/style/redundant_string_escape.rb:160:in `delimiter?'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/style/redundant_string_escape.rb:108:in `single_quoted?'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/style/redundant_string_escape.rb:101:in `interpolation_not_enabled?'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/style/redundant_string_escape.rb:84:in `allowed_escape?'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/style/redundant_string_escape.rb:49:in `block in on_str'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/mixin/match_range.rb:14:in `block in each_match_range'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/mixin/match_range.rb:14:in `scan'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/mixin/match_range.rb:14:in `each_match_range'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/style/redundant_string_escape.rb:48:in `on_str'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:107:in `public_send'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:107:in `block (2 levels) in trigger_responding_cops'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:171:in `with_cop_error_handling'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:106:in `block in trigger_responding_cops'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:105:in `each'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:105:in `trigger_responding_cops'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:69:in `on_str'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:71:in `on_dsym'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:159:in `block in on_send'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:156:in `each'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:156:in `each_with_index'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:156:in `on_send'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:71:in `on_send'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:20:in `walk'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:87:in `investigate'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/team.rb:156:in `investigate_partial'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/team.rb:91:in `investigate'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:345:in `block in inspect_file'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:344:in `each'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:344:in `flat_map'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:344:in `inspect_file'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:287:in `block in do_inspection_loop'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:321:in `block in iterate_until_no_changes'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:314:in `loop'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:314:in `iterate_until_no_changes'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:283:in `do_inspection_loop'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:164:in `block in file_offenses'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:189:in `file_offense_cache'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:163:in `file_offenses'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:154:in `process_file'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:135:in `block in each_inspected_file'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:134:in `each'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:134:in `reduce'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:134:in `each_inspected_file'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:120:in `inspect_files'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:73:in `run'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cli/command.rb:11:in `run'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cli/environment.rb:18:in `run'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cli.rb:118:in `run_command'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cli.rb:125:in `execute_runners'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cli.rb:51:in `block in run'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cli.rb:77:in `profile_if_needed'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cli.rb:43:in `run'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/exe/rubocop:19:in `block in <top (required)>'
/Users/dana/.rubies/ruby-3.1.2/lib/ruby/3.1.0/benchmark.rb:311:in `realtime'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/exe/rubocop:19:in `<top (required)>'
/Users/dana/.gem/ruby/3.1.2/bin/rubocop:25:in `load'
/Users/dana/.gem/ruby/3.1.2/bin/rubocop:25:in `<main>'
An error occurred while Style/RedundantStringEscape cop was inspecting /Users/dana/Projects/test.rb:4:17.
undefined method `source' for nil:NilClass

          delimiters = [node.loc.begin.source[-1], node.loc.end.source[0]]
                                      ^^^^^^^
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/style/redundant_string_escape.rb:160:in `delimiter?'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/style/redundant_string_escape.rb:108:in `single_quoted?'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/style/redundant_string_escape.rb:101:in `interpolation_not_enabled?'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/style/redundant_string_escape.rb:84:in `allowed_escape?'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/style/redundant_string_escape.rb:49:in `block in on_str'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/mixin/match_range.rb:14:in `block in each_match_range'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/mixin/match_range.rb:14:in `scan'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/mixin/match_range.rb:14:in `each_match_range'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/style/redundant_string_escape.rb:48:in `on_str'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:107:in `public_send'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:107:in `block (2 levels) in trigger_responding_cops'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:171:in `with_cop_error_handling'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:106:in `block in trigger_responding_cops'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:105:in `each'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:105:in `trigger_responding_cops'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:69:in `on_str'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:71:in `on_dsym'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:159:in `block in on_send'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:156:in `each'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:156:in `each_with_index'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:156:in `on_send'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:71:in `on_send'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:20:in `walk'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:87:in `investigate'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/team.rb:156:in `investigate_partial'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cop/team.rb:91:in `investigate'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:345:in `block in inspect_file'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:344:in `each'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:344:in `flat_map'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:344:in `inspect_file'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:287:in `block in do_inspection_loop'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:321:in `block in iterate_until_no_changes'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:314:in `loop'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:314:in `iterate_until_no_changes'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:283:in `do_inspection_loop'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:164:in `block in file_offenses'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:189:in `file_offense_cache'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:163:in `file_offenses'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:154:in `process_file'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:135:in `block in each_inspected_file'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:134:in `each'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:134:in `reduce'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:134:in `each_inspected_file'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:120:in `inspect_files'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/runner.rb:73:in `run'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cli/command.rb:11:in `run'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cli/environment.rb:18:in `run'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cli.rb:118:in `run_command'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cli.rb:125:in `execute_runners'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cli.rb:51:in `block in run'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cli.rb:77:in `profile_if_needed'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/lib/rubocop/cli.rb:43:in `run'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/exe/rubocop:19:in `block in <top (required)>'
/Users/dana/.rubies/ruby-3.1.2/lib/ruby/3.1.0/benchmark.rb:311:in `realtime'
/Users/dana/.gem/ruby/3.1.2/gems/rubocop-1.52.1/exe/rubocop:19:in `<top (required)>'
/Users/dana/.gem/ruby/3.1.2/bin/rubocop:25:in `load'
/Users/dana/.gem/ruby/3.1.2/bin/rubocop:25:in `<main>'
.

1 file inspected, no offenses detected

2 errors occurred:
An error occurred while Style/RedundantStringEscape cop was inspecting /Users/dana/Projects/test.rb:4:7.
An error occurred while Style/RedundantStringEscape cop was inspecting /Users/dana/Projects/test.rb:4:17.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.52.1 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.1.2) [x86_64-darwin20]
Finished in 0.23604999999224674 seconds
```

## Steps to reproduce the problem

create a file named test.rb with:

```ruby
value = 1
:"\"#{value}\""
```

and run

$ rubocop -a test.rb --only Style/RedundantStringEscape

## RuboCop version

```
1.52.1
```
